Lab/app/src/main/java/com/elysium/lab/MainActivity.java

```diff
@@ -0,0 +1,54 @@
+package com.elysium.lab;
+
+import android.animation.LayoutTransition;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.animation.Animation;
+import android.view.animation.AnimationUtils;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+public class MainActivity extends AppCompatActivity {
+
+    private EditText mEditText;
+    private TextView mAnimation;
+    private ViewGroup mViewGroup;
+    private Button mButton;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mEditText = (EditText) findViewById(R.id.edit_text);
+        mAnimation = (TextView) findViewById(R.id.animation);
+        mButton = (Button) findViewById(R.id.button);
+
+        LayoutTransition layoutTransition = new LayoutTransition();
+        layoutTransition.enableTransitionType(LayoutTransition.CHANGING);
```

Comment:
Since you put android:animateLayoutChanges="true" in the XML for the LinearLayout, you actually don't need to also set up the layout transition animation here. That one line in XML accomplishes the same thing.